File: scripts/clean_paths.py
from __future__ import annotations

import pathlib
import shutil
import sys


def main() -> int:
    if len(sys.argv) < 2:
        raise SystemExit("Usage: python scripts/clean_paths.py <path> [<path> ...]")

    cwd = pathlib.Path.cwd()

    for target in sys.argv[1:]:
        resolved = cwd / target
        if resolved.is_dir():
            shutil.rmtree(resolved, ignore_errors=True)
        elif resolved.exists():
            resolved.unlink()

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: scripts/clean-paths.ts
import path from "node:path";
import { pathToFileURL } from "node:url";
import { rmSync } from "node:fs";

export function cleanPaths(cwd: string, targets: readonly string[]): void {
  if (targets.length === 0) {
    throw new Error("Usage: pnpm exec tsx ./scripts/clean-paths.ts <path> [<path> ...]");
  }

  for (const target of targets) {
    rmSync(path.join(cwd, target), {
      force: true,
      recursive: true,
      maxRetries: 5,
      retryDelay: 200
    });
  }
}

function main(): void {
  cleanPaths(process.cwd(), process.argv.slice(2));
}

if (process.argv[1] && import.meta.url === pathToFileURL(process.argv[1]).href) {
  main();
}
